Feature/cicd

sridharvikram opened the pull request:
I have added cicd pipeline code and readme doc to set it up for migrating agent under examples folder. I would like contribute this to SCRAPI library. Please review the changes

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/export/export.py:
```diff
@@ -0,0 +1,99 @@
+
+
+
+def agentToGcs(
```
Need to follow snake_case python naming standards.

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/export/export.py:
```diff
@@ -0,0 +1,99 @@
+    gs_loc,
+    source_agent_name,
+    bucket,
+    impactedVersionIds):
```
prefer snake_case

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/export/flowimpacted.py:
```diff
@@ -0,0 +1,108 @@
+
+from typing import Dict, List
+
+class impacted:
```
class names should be capitalized in Python.
2 spaces before the class starts

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/export/flowimpacted.py:
```diff
@@ -0,0 +1,108 @@
+
+
+
+    def checkFlow(
```
snake_case

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/readme.md:
```diff
@@ -0,0 +1,331 @@
+4.  Now you can come back to cloud build console and build history tab
+    and approve the build that is waiting for your approval and you can
+    see that it will deploy the agent in prod post approval
+    !media/image6.png
```
this image is not loading in the markdown file

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/readme.md:
```diff
@@ -0,0 +1,331 @@
+    see that it will deploy the agent in prod post approval
+    !media/image6.png
+
+## Cavet
```
sp: `Caveat`

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/shared/deployment.py:
```diff
@@ -0,0 +1,173 @@
+            setattr(self, key, value)     
+
+    #[1] import to destination project
+    def importAgent(self,webhookenv):
```
prefer snake_case

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/shared/enVsOtherLang.py:
```diff
@@ -0,0 +1,149 @@
+      return response
+
+
+def enVslang(agent_id,flows,lang):
```
prefer snake_case

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/shared/fullfillmentHelper.py:
```diff
@@ -0,0 +1,68 @@
+
+
+def getEntryff(page=None,page_id=None,language_code='en'):
```
prefer snake_case

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/shared/testcaserun.py:
```diff
@@ -0,0 +1,100 @@
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+class runtestcases:
```
capitalize Class names

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix commented on examples/dfcx_agent_cicd/cicd_code/shared/testcaserun.py:
```diff
@@ -0,0 +1,100 @@
+        else:
+            self.env_id=None
+
+    def triggerTestcase(
```
snake_case

sridharvikram replied:
Fixed in my latest push to my branch

kmaphoenix reviewed (changes requested):
Hi @sridharvikram there are several changes I've tagged in the files that are Python style updates that need to be made.

In addition, please run `pylint` locally to see all of the linter errors that need to be corrected.
You can also view the latest pylint run directly in the repo here: 
https://github.com/GoogleCloudPlatform/dfcx-scrapi/actions/runs/9990092140/job/27632420520?pr=196

sridharvikram commented:
@kmaphoenix I have addressed all the lint issues you have mentioned and also fixed most of the lint errors that i ran in my IDE. Please re review it and do the needful.

kmaphoenix commented:
Hi @sridharvikram , did you perform a `git rebase` with `main` branch?

It's showing many edits to files that are not related to your CICD code and should not have been modified. 

If you did not perform a rebase, you should do this first. 
If you're not familiar with running a rebase, you can consider starting a new branch from the latest main branch. 

Otherwise it makes it very difficult to review all of your code.

kmaphoenix commented:
@sridharvikram yeah after looking at the branch some more there are many merge conflicts. 

Best course of action:
- Start a new branch from `main`
- Add your changes for CICD code
- Submit PR

Otherwise we're going to have a lot of back and forth to get this fixed and you'll need to fix all of the merge conflicts locally.

sridharvikram commented:
@kmaphoenix I have created this new PR https://github.com/GoogleCloudPlatform/dfcx-scrapi/pull/239 as suggested. Hence closing this one. Thanks
